src/parthenon_array_generic.hpp

```diff
@@ -339,6 +312,14 @@ inline bool UseSameResource(const PA &pa1, const PA &pa2) {
 // assumed ParArrayGeneric = Kokkos::View does not need to be changed
 namespace Kokkos {
 
+// JMM: for some reason this works better than Slice. And it seems
+// like we're doing evil Kokkos namespace overloads anyway so...
+template <class U, class SU, typename... Args>
+inline auto subview(const parthenon::ParArrayGeneric<U, SU> &arr, Args... args) {
+  auto v = Kokkos::subview(static_cast<U>(arr), std::forward<Args>(args)...);
+  return parthenon::ParArrayGeneric<decltype(v), SU>(v, arr);
+}
+
```

Comment:
Is this actually used anywhere right now?
Otherwise, we could also just skip this and encourage using `Kokkos::subview` directly (as already done in various places of the code).

Reply:
It is used---in the updated prolongation/restriction logic, as well as in `refinement-in-one`. You just can't tell what is what because of the evil overloading